app/src/main/java/com/cornellappdev/android/eatery/util/AppStorePopupRepository.kt

```diff
@@ -0,0 +1,66 @@
+package com.cornellappdev.android.eatery.util
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import com.cornellappdev.android.eatery.data.repositories.PopupDataRepository
+import com.cornellappdev.android.eatery.data.repositories.UserPreferencesRepository
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
+import dagger.hilt.android.EntryPointAccessors
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import java.time.Duration
+import java.time.LocalDate
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AppStorePopupRepository @Inject constructor(
+    val userPreferencesRepository: UserPreferencesRepository,
+    private val popupDataRepository: PopupDataRepository,
+    private val appScope: CoroutineScope,
+) {
+    private val popupEventFlow = MutableStateFlow(false)
+    val popupShowing = popupEventFlow.asStateFlow()
+
+    fun requestRatingPopup() {
```

Comment:
I was a bit confused looking at this function until I realized that it uses `popupDataRepository` to determine if it *should* show the dialog. I suppose I thought that this would cause a pop up to show whenever it's called.

Let's add a spec comment to clarify that this only will indeed cause the show to happen if it is appropriate to do so (user hasn't been shown recently)

Reply:
good catch! Will add a spec 👍